src/corecel/sys/KernelDiagnostics.hh

```diff
@@ -126,6 +135,7 @@ void KernelDiagnostics::launch(key_type key, unsigned int num_threads)
 {
     CELER_EXPECT(key < this->size());
     value_type& diag = values_[key.get()];
+    // TODO: the following diagnostics should be atomic in MT environment
```

Comment:
I left this as a to-do because I'm not sure whether the ramifications of changing this to an atomic are worth it. (I don't know how much it'll hurt kernel launch performance, nor have I played around with getting the values written out to JSON.)

Reply:
When is it this launch function used?  I.e. all the time or when 'diagnostic' is on?

When are `num_launches` and `max_num_threads` used?  I.e. can we assigned then "only" with the `     if (CELERITAS_DEBUG)` section?

Reply:
All the time: we currently increment the counter diagnostics at every kernel call. These are only used for diagnostics in the output, which I suppose we could indeed turn off for production runs?